packages/rolldown/src/cli/index.ts

```diff
@@ -1,55 +1,34 @@
 import process from 'node:process'
-import parseArgs from 'mri'
-import { defineCommand, runMain, showUsage } from 'citty'
+import { parseArgs } from 'node:util'
+import { mapValues, omit } from 'remeda'
 import { bundle } from './commands/bundle'
-import {
-  version,
-  description,
-} from '../../package.json' assert { type: 'json' }
+import { showHelp } from './commands/help'
 import { DEFAULT_CONFIG_FILENAME } from './constants'
+import { CLI_OPTIONS } from './options'
+import { logger } from './utils'
 
-interface ParsedArgs {
-  config?: string | true
-  c?: string | true
-  // `citty` intercept the help option, so we don't need to deal with it
-  // help?: boolean
-  // h?: boolean
-}
+async function main() {
+  const { values } = parseArgs({
+    options: mapValues(CLI_OPTIONS, omit(['description'])),
+    // We need to support both `rolldown -c` and `rolldown -c rolldown.config.js`,
+    // the value of the option could be either a boolean or a string in this case,
+    // so `strict` needs to be set to `false`
+    strict: false,
```

Comment:
Indeed, it is true. By the way, I wonder if we should specify the `rolldown.config.js` file or move it to the constants file (as it may be considered hardcoding if I remove the “constants” file.).

Additionally, I do not believe the automatically ignored parameter is elegant. Once the API becomes stable and independent, I think we should look into a better way to separate the default configuration or the configuration file that’s been specified :thinking:.

Reply:
> Indeed, it should be `false`. By the way, I wonder if we should specify the `rolldown.config.js` file or move it to the constants file (as it may be considered hardcoding if I remove the “constants” file.).

It should be fine to keep it in constants.ts, as there might be some default values and so on for the CLI in the future.

> Additionally, I do not believe the automatically ignored parameter is elegant. Once the API becomes stable and independent, I think we should look into a better way to separate the default configuration or the configuration file that’s been specified 🤔.

Perhaps when the user specifies both the config option and other options, we could log a message to inform them that we will ignore the other options?

Reply:
> It should be fine to keep it in constants.ts, as there might be some default values and so on for the CLI in the future.

Ok.

> Perhaps when the user specifies both the config option and other options, we could log a message to inform them that we will ignore the other options?

I'll do it in #1963.